metadata/content.go

```diff
@@ -318,12 +318,34 @@ func (cs *contentStore) Writer(ctx context.Context, ref string, size int64, expe
 	cs.l.RLock()
 	defer cs.l.RUnlock()
 
-	var w content.Writer
+	var (
+		w         content.Writer
+		existsErr error
+	)
 	if err := update(ctx, cs.db, func(tx *bolt.Tx) error {
 		if expected != "" {
+			// check the backend to see if we already have this content
+			// that may be pinned in another namespace.
+			if info, err := cs.Store.Info(ctx, expected); err != nil {
```

Comment:
@stevvooe I don't see how this would ever work.  It would check the namespace in the context and that ns won't have the content, even if it already exists.

Reply:
This is hitting the backend storage, which isn't namespaced.